applications/OptimizationApplication/python_scripts/controls/shape/vertex_morphing_shape_control.py

```diff
@@ -0,0 +1,363 @@
+            raise RuntimeError(f"Updates for the required element container not found for control \"{self.GetName()}\". [ required model part name: {self.filter_model_part.FullName()}, given model part name: {new_control_field.GetModelPart().FullName()} ]")
+        if KratosOA.ExpressionUtils.NormL2(self.control_field - new_control_field) > 1e-15:
+            with TimeLogger(self.__class__.__name__, f"Updating {self.GetName()}...", f"Finished updating of {self.GetName()}.",False):
+                # update the control SHAPE field
+                control_update = new_control_field - self.control_field
+                self.control_field = new_control_field
+                # now update the physical field
+                self._UpdateAndOutputFields(control_update)
+                # update explicit filter
+                if not self.is_filter_implicit:
+                    self.filter.Update()
+                return True
+        return False
+
+    def _UpdateMesh(self, shape_update) -> None:
+        if self.has_mesh_motion:
+            mm_computing_model_part = self._mesh_moving_analysis._GetSolver().GetComputingModelPart()
+            time_before_update = mm_computing_model_part.ProcessInfo.GetValue(Kratos.TIME)
+            step_before_update = mm_computing_model_part.ProcessInfo.GetValue(Kratos.STEP)
+            delta_time_before_update = mm_computing_model_part.ProcessInfo.GetValue(Kratos.DELTA_TIME)
+
+            # Reset step/time iterators such that they match the current iteration after calling RunSolutionLoop (which internally calls CloneTimeStep)
+            mm_computing_model_part.ProcessInfo.SetValue(Kratos.STEP, step_before_update-1)
+            mm_computing_model_part.ProcessInfo.SetValue(Kratos.TIME, time_before_update-1)
+            mm_computing_model_part.ProcessInfo.SetValue(Kratos.DELTA_TIME, 0)
+
+            # assign the boundary_displacements
+            Kratos.Expression.VariableExpressionIO.Write(shape_update, Kratos.MESH_DISPLACEMENT, True)
+
+            # solve for the volume mesh displacements
+            if not self._mesh_moving_analysis.time < self._mesh_moving_analysis.end_time:
+                self._mesh_moving_analysis.end_time += 1
+            self._mesh_moving_analysis.RunSolutionLoop()
+
+            # update the coords
+            mesh_disp_field = Kratos.Expression.NodalExpression(mm_computing_model_part)
+            Kratos.Expression.LiteralExpressionIO.SetData(mesh_disp_field, [0,0,0])
+            Kratos.Expression.VariableExpressionIO.Read(mesh_disp_field, Kratos.MESH_DISPLACEMENT, True)
+            physical_shape_field = Kratos.Expression.NodalExpression(mm_computing_model_part)
+            Kratos.Expression.NodalPositionExpressionIO.Read(physical_shape_field, Kratos.Configuration.Initial)
+            Kratos.Expression.NodalPositionExpressionIO.Write(mesh_disp_field + physical_shape_field, Kratos.Configuration.Initial)
```

Comment:
why do you  need to rewrite Initial? I guess they should always stay as Initial coordinates. like you can always call current and compute new_current as current + mesh_displacements. For that you don't need to rewrite Initial coords.

Reply:
The point is that almost all elements of Kratos use the initial configuration/coordinates to compute stiffness,... . Plus, implicit and explicit filters as well as the sensitivity analysis are all using the initial configuration for the computation, therefore we need to update the initial as well as current

Reply:
ok, if it is needed for promal analysis, it is fine to update.